dev-tools/observability/validate-observability-contract.py

```
@@ -68,17 +147,425 @@ def _github_anchors_for_markdown(path: Path) -> set[str]:
+    if not ranges:
+        standalone_range = _standalone_rule_sequence_range(lines)
+        if standalone_range is not None:
+            ranges.append(standalone_range)
+
+    for start, end, sequence_indent in ranges:
+        starts = [
+            index
+            for index in range(start, end)
+            if _is_sequence_item(lines[index], sequence_indent)
+        ]
+        for entry_start, entry_end in zip(starts, [*starts[1:], end]):
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)